astropy/table/table.py

```diff
@@ -1778,89 +1778,55 @@ def _make_index_row_display_table(self, index_row_name):
         else:
             return self
 
-    @deprecated(
-        "6.1",
-        message="""show_in_notebook() is deprecated as of 6.1 and to create
-         interactive tables it is recommended to use dedicated tools like:
-         - https://github.com/bloomberg/ipydatagrid
-         - https://docs.bokeh.org/en/latest/docs/user_guide/interaction/widgets.html#datatable
-         - https://dash.plotly.com/datatable""",
-    )
-    def show_in_notebook(
-        self,
-        tableid=None,
-        css=None,
-        display_length=50,
-        table_class="astropy-default",
-        show_row_index="idx",
-    ):
+    def show_in_notebook(self, *, backend="classic", **kwargs):
```

Comment:
This is a breaking API change anyway, so let's go all the way and make the non-deprecated ipydatagrid backend be the default.

```suggestion
    def show_in_notebook(self, *, backend="ipydatagrid", **kwargs):
```

Reply:
Hmm I thought we wanted the defaults to be backward compatible so people won't be surprised?